integrations/signin_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integrations
+
+import (
+	"net/http"
+	"strings"
+	"testing"
+
+	"code.gitea.io/gitea/models"
+
+	"github.com/Unknwon/i18n"
+	"github.com/stretchr/testify/assert"
+)
+
+func testLoginFailed(t *testing.T, username, password, message string) {
+	session := emptyTestSession(t)
+	req := NewRequestWithValues(t, "POST", "/user/login", map[string]string{
+		"_csrf":     GetCSRF(t, session, "/user/login"),
+		"user_name": username,
+		"password":  password,
+	})
+	resp := session.MakeRequest(t, req, http.StatusOK)
```

Comment:
@ethantkoenig first of all to prevent CSRF attack ;)
http://pico.ninja/gitea-login.html - try entering your try.gitea.io username/password :P (don't worry it's not logged just to demonstrate idea)